pkg/templates/k8sresource_funcs_test.go

```diff
@@ -118,3 +119,186 @@ func TestFromConfigMap(t *testing.T) {
+		}, // error : restricted input namespace
+	}
+
+	for _, test := range testcases {
+		resolver, err := NewResolver(&k8sClient, k8sConfig, Config{LookupNamespace: test.lookupNamespace})
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+
+		val, err := resolver.copySecretDataJSON(test.inputNs, test.inputSecretName)
+
+		if err != nil {
+			if test.expectedErr == nil {
+				t.Fatalf(err.Error())
+			}
+
+			if !strings.EqualFold(test.expectedErr.Error(), err.Error()) {
+				t.Fatalf("expected err: %s got err: %s", test.expectedErr, err)
+			}
+		} else if val[test.inputKey] != base64encode(test.expectedResult) {
+			t.Fatalf("expected : %s , got : %s", base64encode(test.expectedResult), val)
+		}
+	}
+}
+
+func TestCopySecretDataProtected(t *testing.T) {
+	t.Parallel()
+
+	testcases := []struct {
+		inputNs         string
+		inputSecretName string
+		inputKey        string
+		lookupNamespace string
+		expectedResult  string
+		expectedErr     error
+	}{
+		{
+			"testns", "testsecret", "secretkey1", "",
+			"$ocm_encrypted:c6PNhsEfbM9NRUqeJ+HbcECCyVdFnRbLdd+n8r1fS9M=", nil,
+		}, // green-path test
+		{
+			"testns", "testsecret", "secretkey2", "",
+			"$ocm_encrypted:VlXOhKuKGoHimHAYlQ2xz5EBw7mriqtt7fEP5ShP5cw=", nil,
+		}, // green-path test
+		{
+			"testns", "testsecret", "secretkey2", "testns",
+			"$ocm_encrypted:VlXOhKuKGoHimHAYlQ2xz5EBw7mriqtt7fEP5ShP5cw=", nil,
+		}, // green-path test
+		{
+			"testns",
+			"idontexist",
+			"secretkey1",
+			"",
+			"secretkey2Val",
+			errors.New(`failed to get the secret idontexist from testns: secrets "idontexist" not found`),
+		}, // error : nonexistant secret
+		{"testns", "testsecret", "blah", "", "", nil}, // error : nonexistant key
+		{
+			"testns",
+			"testsecret",
+			"secretkey2",
+			"policies-ns",
+			"",
+			errors.New("the namespace argument passed to copySecretData is restricted to policies-ns"),
+		}, // error : restricted input namespace
+	}
+
+	for _, test := range testcases {
+		iv := bytes.Repeat([]byte{byte('I')}, IVSize)
+
+		resolver, err := NewResolver(&k8sClient, k8sConfig, Config{
+			EncryptionConfig: EncryptionConfig{
+				AESKey:               bytes.Repeat([]byte{byte('A')}, 256/8),
+				EncryptionEnabled:    true,
+				InitializationVector: iv,
+			},
+			LookupNamespace: test.lookupNamespace,
+		})
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+
+		val, err := resolver.copySecretDataProtected(test.inputNs, test.inputSecretName)
+
+		if err != nil {
+			if test.expectedErr == nil {
+				t.Fatalf(err.Error())
+			}
+
+			if !strings.EqualFold(test.expectedErr.Error(), err.Error()) {
+				t.Fatalf("expected err: %s got err: %s", test.expectedErr, err)
+			}
+		} else if !strings.Contains(val, test.expectedResult) {
+			t.Fatalf("expected to contain: %s got val: %s", test.expectedResult, val)
+		}
+	}
+}
+
+func TestCopyConfigMapData(t *testing.T) {
+	t.Parallel()
+
+	testcases := []struct {
```

Comment:
Could you refactor your tests to account for #81?

Reply:
I'd like to consider the refactor of the tests as a separate PR not associated with this initial work if possible.  I haven't had a chance to look into this yet.

Reply:
Sure, no problem.